### ISSUE TITLE
TypeError: Cannot read property 'name' of undefined

### ISSUE THREAD

**Issue by danvk** [jens-ox/eslint-plugin-material-ui-unused-classes#11]
I got this error when running this plugin in my project:

```
$ eslint '**/*.ts{,x}' --max-warnings=0

Oops! Something went wrong! :(

ESLint: 8.32.0

TypeError: Cannot read property 'name' of undefined
Occurred while linting /project/.../OverflowText.tsx:14
Rule: "mui-unused-classes/unused-classes"
    at CallExpression (/project/node_modules/eslint-plugin-mui-unused-classes/rule.js:39:37)
    at ruleErrorHandler (/project/node_modules/eslint/lib/linter/linter.js:1118:28)
    at /project/node_modules/eslint/lib/linter/safe-emitter.js:45:58
    at Array.forEach (<anonymous>)
    at Object.emit (/project/node_modules/eslint/lib/linter/safe-emitter.js:45:38)
    at NodeEventGenerator.applySelector (/project/node_modules/eslint/lib/linter/node-event-generator.js:297:26)
    at NodeEventGenerator.applySelectors (/project/node_modules/eslint/lib/linter/node-event-generator.js:326:22)
    at NodeEventGenerator.enterNode (/project/node_modules/eslint/lib/linter/node-event-generator.js:340:14)
    at CodePathAnalyzer.enterNode (/project/node_modules/eslint/lib/linter/code-path-analysis/code-path-analyzer.js:795:23)
    at /project/node_modules/eslint/lib/linter/linter.js:1153:32
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

This was the file that triggered it:

```ts
import Box, {BoxProps} from '@material-ui/core/Box';
import {createStyles, makeStyles} from '@material-ui/core/styles';
import Typography, {TypographyProps} from '@material-ui/core/Typography';
import React from 'react';

export interface Props {
  maxLines: number;
  boxProps?: BoxProps;
  typographyProps?: TypographyProps;
  children: React.ReactNode;
}

export const OverflowText = ({maxLines, boxProps = {}, typographyProps = {}, children}: Props) => {
  const classes = makeStyles(() =>
    createStyles({
      overflowText: {
        display: '-webkit-box',
        '-webkit-line-clamp': maxLines,
        '-webkit-box-orient': 'vertical',
        overflow: 'hidden',
      },
    }),
  )();

  return (
    <Box className={classes.overflowText} {...boxProps}>
      <Typography {...typographyProps}>{children}</Typography>
    </Box>
  );
};
```

I was able to work around the issue by wrapping this line in an `if` check:

https://github.com/jens-ox/eslint-plugin-material-ui-unused-classes/blob/857559ac4cb71db5d54904e3137e8e1ec70c2950/rule.js#L39

```ts
          if (!node.parent.id) {
            return;
          }
          hookName = node.parent.id.name
```

**Comment by jens-ox** [comment]
Hi @danvk! I'm a bit confused by this syntax:

```js
const classes = makeStyles(() =>
  createStyles({
    overflowText: {
      overflow: 'hidden',
    },
  }),
)()
```

I didn't know that one can do that! 😄

**Comment by jens-ox** [comment]
I patched this for now (in 2.0.2) - it won't detect unused classes in this case, but at least it won't throw.
